### PR TITLE
8283339: TypeError: undefined is not an Object after JDK-8240299

### PR DESCRIPTION
**Pull request by cushon**
Contributed-by: Nicholas Burbaum <nburbaum@google.com>

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283339](https://bugs.openjdk.java.net/browse/JDK-8283339): TypeError: undefined is not an Object after JDK-8240299

### Reviewers
 * [Attila Szegedi](https://openjdk.java.net/census#attila) (@szegedi - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/nashorn pull/17/head:pull/17` \
`$ git checkout pull/17`

Update a local copy of the PR: \
`$ git checkout pull/17` \
`$ git pull https://git.openjdk.java.net/nashorn pull/17/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17`

View PR using the GUI difftool: \
`$ git pr show -t 17`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/nashorn/pull/17.diff">https://git.openjdk.java.net/nashorn/pull/17.diff</a>

</details>

### CODE REVIEW

**Comment by szegedi** [comment]
I see that Nicholas has no OCA on file, but I'll presume that we can proceed without him signing one since:

-  we can consider him a Participant instead of Contributor as per [OpenJDK Bylaws](https://openjdk.java.net/bylaws#participant) as this I believe qualifies as a "small patch", and
- he would fall under OCA Google signed (visible on the [OCAs submitted prior to 2021](https://oca.opensource.oracle.com/?ojr=old-contrib-list#g) page.)

File: test/nashorn/script/basic/JDK-8283339.js.EXPECTED
Comment (szegedi) [inline test/nashorn/script/basic/JDK-8283339.js.EXPECTED:1]: Please add a newline at the end of file. It's weird the test passes since `print(a)` on line 34 in the test does emit a newline.

**Review by szegedi — CHANGES_REQUESTED** [review]

File: test/nashorn/script/basic/JDK-8283339.js.EXPECTED
Comment (cushon) [reply test/nashorn/script/basic/JDK-8283339.js.EXPECTED]: Done, thanks for the review!

**Review by szegedi — APPROVED** [review]

**Comment by cushon** [comment]
I pushed one more commit to fix the newly-added file to have Google copyright (which is standard, see e.g. [this thread](https://mail.openjdk.java.net/pipermail/compiler-dev/2016-April/010138.html)).

**Review by szegedi — APPROVED** [review]

**Comment by cushon** [comment]
@szegedi thanks for the review! I don't have committer status, so when you're ready for this to be merged I think you'll need to sponsor it before it can be merged by commenting `/sponsor`

**Comment by szegedi** [comment]
I believe you need to re-integrate after the last commit, and then I can sponsor it. You can see that the bot removed the "sponsor" label after you added the commit after integration.

**Comment by szegedi** [comment]
Hey @cushon, just wanted to let you know that this is now released in Nashorn 15.4. Thanks again for the contribution!

**Pull request closed without merging**
